cmdstanpy/utils/logging.py

```diff
@@ -1,25 +1,62 @@
 """
 CmdStanPy logging
 """
+
 import functools
 import logging
+import types
+from contextlib import AbstractContextManager
+from typing import Optional, Type
 
 
 @functools.lru_cache(maxsize=None)
 def get_logger() -> logging.Logger:
     """cmdstanpy logger"""
-    logger = logging.getLogger('cmdstanpy')
-    if len(logger.handlers) == 0:
+    logger = logging.getLogger("cmdstanpy")
+    if not logger.hasHandlers():
         # send all messages to handlers
         logger.setLevel(logging.DEBUG)
         # add a default handler to the logger to INFO and higher
         handler = logging.StreamHandler()
         handler.setLevel(logging.INFO)
         handler.setFormatter(
             logging.Formatter(
-                '%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                "%(asctime)s - %(name)s - %(levelname)s - %(message)s",
                 "%H:%M:%S",
             )
         )
         logger.addHandler(handler)
     return logger
+
+
+class ToggleLogging(AbstractContextManager):
+    def __init__(self, disable: bool) -> None:
+        self.disable = disable
+        self.logger = get_logger()
+        self.prev_state = self.logger.disabled
+        self.logger.disabled = self.disable
+
+    def __repr__(self) -> str:
+        return ""
+
+    def __enter__(self) -> "ToggleLogging":
+        self.logger.disabled = self.disable
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[types.TracebackType],
```

Comment:
[optional] I realize these are just here to make typing happy, since we don't need them can we get away with `*args, **kwargs`? That way we don't need to import quite as much just for typing purposes